packages/html-reporter/src/labels.tsx

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+import { navigate } from './links';
+import type { TestCaseSummary } from './types';
+
+/**
+ * Copyright (c) Microsoft Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+export const Labels: React.FC<React.PropsWithChildren<{
+  testCase: TestCaseSummary,
+  style?: React.CSSProperties,
+}>> = ({ testCase, style }) => {
+
+  const labels = React.useMemo(() => {
+    return matchTags(testCase.title).sort((a, b) => a.localeCompare(b));
+  }, [testCase]);
+
+  if (!labels?.length)
+    return null;
+
+  return labels && <div className='labels' style={{ ...style }}>
+    {labels && labels.map(tag => {
+      return <Label key={tag} tag={tag} />;
+    })}
+  </div>;
+};
+
+export const Label: React.FC<React.PropsWithChildren<{
+  tag: string,
+}>> = ({ tag }) => {
+  const encoded = encodeURIComponent(tag);
+  const value = tag === encoded ? tag : `"${encoded.replace(/%22/g, '%5C%22')}"`;
+
+  const onClickHandle = (e: React.MouseEvent, tag: string) => {
+    e.preventDefault();
+
+    const searchParams = new URLSearchParams(window.location.hash.slice(1));
+    const q = searchParams.get('q') || '';
+
+    if (searchParams.has('testId')){
```

Comment:
Looks like we have two different types of links: simple links like `ProjectLink` that you use in tests and in this branch and a smart ctrl+clickable link. Let's split them?